src/pyggdrasil/tree_inference/_mcmc.py

```diff
@@ -284,6 +290,13 @@ def _mcmc_kernel(
     else:
         proposal, log_q_diff = _swap_subtrees_proposal(key, tree)
 
+    # TODO:Consider replacing the above condition with
+    # condlist = [move_type == 0, move_type == 1]
+    # choicelist = [_prune_and_reattach_proposal(key, tree),
+    #               _swap_node_labels_proposal(key, tree)]
+    # default = _swap_subtrees_proposal(key, tree)
+    # proposal, log_q_diff = jnp.select(condlist, choicelist, default)
+
```

Comment:
Doubt: In line-comment, I have a `jit` compatible version of these `if, elif, else` conditions. This is a relict from my attempts to use use `jax.lax.while_loop` in the main mcmc loop.  I quickly realized that a bunch of modifications are needed, deeper down in 'get_descendants' and so on. - Fixing this to use jax properly will be in the next sprint I suppose?

Reply:
For if/else I often use [jax.lax.cond](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.cond.html). But don't worry about making it JAX-compatible now. I think with the current implementation we can already do some cool research :wink: